### PR TITLE
Move the group database backend fully to query builder

### PR DESCRIPTION
**Pull request by nickvergessen**
So we have a cleaner basement for up coming work

### CODE REVIEW

File: lib/private/Group/Database.php [hunk 1]
```diff
@@ -237,19 +239,27 @@ public function getUserGroups( $uid ) {
...
 	public function getGroups($search = '', $limit = null, $offset = null) {
-		$parameters = [];
-		$searchLike = '';
+		$query = $this->dbConn->getQueryBuilder();
+		$query->select('gid')
+			->from('groups')
+			->orderBy('gid', 'ASC');
+
 		if ($search !== '') {
-			$parameters[] = '%' . $search . '%';
```
Comment (nickvergessen) [inline lib/private/Group/Database.php]: Actually this was missing `escapeLikeParameter` so if you search for `%` or `*` it would go ham.

**Comment by rullzer** [comment]
CI is not happy

**Review by blizzz — APPROVED** [review]

**Comment by nickvergessen** [comment]
Yeah, right. That's why it didnt use DI...

**Comment by MorrisJobke** [comment]
😢 

```
User <admin> could not be created.
```

**Comment by nickvergessen** [comment]
okay, back to `fixDI()` :crying_cat_face:

**Comment by MorrisJobke** [comment]
> okay, back to fixDI() 😿

😢

**Comment by rullzer** [comment]
I rebased to see if see can get a bit greener.

**Review by rullzer — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Pull request merged**